### PR TITLE
[AutoMM] Fix hyperparameters consistency between init() and fit()

### PR DESCRIPTION
**Pull request by zhiqiangdon**
*Issue #, if available:*
#2978 

*Description of changes:*

- Make `hyperprameters` consistent between predictor init() and fit().
- Make `hyperparameters` consistent between matcher init() and fit().
- Replace `config.optimization.val_metric` with `validation_metric` in init(), which is to reduce hacking and simplify the logic.
- Replace `init_scracth` with `pretrained` in init() to align with AutoMM motivation and other internal APIs. AutoMM is mainly to finetune pretrained models. `pretrained` is more intuitive for users compared to `init_scratch`. Some internal APIs already used `pretrained`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/presets.py [hunk 1]
```diff
@@ -375,7 +375,7 @@ def object_detection(presets: str = DEFAULT):
         "optimization.top_k_average_method": "best",
         "optimization.warmup_steps": 0.0,
         "optimization.patience": 10,
-        "optimization.val_metric": "map",
+        # "optimization.val_metric": "map",
```
Comment (tonyhoo) [inline multimodal/src/autogluon/multimodal/presets.py:378]: why we comment out this?

File: multimodal/src/autogluon/multimodal/matcher.py [hunk 2]
```diff
@@ -285,6 +279,66 @@ def set_verbosity(self, verbosity: int):
         self._verbosity = verbosity
         set_logger_verbosity(verbosity, logger=logger)
 
+    def _init_pretrained(self):
+
+        if self._config is None:
+            # split out the hyperparameters whose values are complex objects
+            hyperparameters, advanced_hyperparameters = split_hyperparameters(self._hyperparameters)
```
[the remark this answers is not in the document]
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/matcher.py:286]: advanced_hyperparameters are to support more image augmentations from torchvision transforms (https://pytorch.org/vision/stable/transforms.html#geometry). In this case, the values of hyperparameters are callable objects instead of strings or numbers.

File: multimodal/src/autogluon/multimodal/presets.py (hunk 1 above)
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/presets.py:378]: Because this PR changes the design of customizing validation metric. `optimization.val_metric` has been removed from the default config.

File: multimodal/src/autogluon/multimodal/matcher.py [hunk 3]
```diff
@@ -126,6 +126,8 @@ def __init__(
         verbosity: Optional[int] = 3,
         warn_if_exist: Optional[bool] = True,
         enable_progress_bar: Optional[bool] = None,
+        pretrained: Optional[bool] = True,
```
[the remark this answers is not in the document]
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/matcher.py:129]: `MultiModalMatcher` is not exposed to users. We can add the docstring later.

File: multimodal/src/autogluon/multimodal/presets.py (hunk 1 above)
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/presets.py]: Removed the commented line.

File: multimodal/src/autogluon/multimodal/predictor.py [hunk 4]
```diff
@@ -493,14 +487,17 @@ def match_label(self):
 
     @property
     def problem_type(self):
```
[the remark this answers is not in the document]
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/predictor.py:489]: Updated to set predictor._matcher._pipeline to predictor._problem_type (https://github.com/autogluon/autogluon/pull/3279/files#diff-a73844176f6eae3b4538467b3565e3bdbf4d88997c3c670173bbb44e1a55b7a3R2700) when loading a saved matcher.
